raw_linux.go

```diff
@@ -279,6 +279,34 @@ func (p *packetConn) SetBPF(filter []bpf.RawInstruction) error {
 	)
 }
 
+// SetPromisc enables/disables interface promiscuous mode through syscall
+// when thread exits interface returns to previous state
+// 0 == disabled, 1 == enabled
+type packet_mreq struct {
```

Comment:
I'll see if I can submit a CL to get this data structure added to https://godoc.org/golang.org/x/sys/unix, so we can use the canonical one upstream.